src/connection/session.ts

```diff
@@ -0,0 +1,73 @@
+import { createConnection } from 'net';
+import { type CompactString, type NullableString } from '../protocol/primitives';
+import { ApiVersionsRequestBuilder } from '../protocol/requests/api-versions/api-versions-request-builder';
+import { type RequestBuilder } from '../protocol/requests/request-builder';
+import { Connection } from './connection';
+
+type SupportedApiVersions = {
+  [apiKey: number]: {
+    min: number;
+    max: number;
+  };
+};
+
+export class Session {
```

Comment:
Responsible for creating tcp connection and doing all initial handshakes on creation